modules/nwtc-library/src/NWTC_IO.f90

```diff
@@ -4375,12 +4247,11 @@ SUBROUTINE Cleanup (  )
             ! Deallocate the linked list of file names.
 
           CurrFile => FirstFile
-          NextFile => CurrFile%Next
-          DO
+          DO WHILE(ASSOCIATED(currFile))
+              NextFile => CurrFile%Next
```

Comment:
I am hopeful that this fixes some of the weird behavior that was sometimes causing regression tests to fail on file reads when the OpenFAST input file reading was changed to use this routine!

Reply:
I hope it improves the stability. This is definitely in the realm of undefined behavior, so it could have been doing anything when CurrFile was not associated